### PR TITLE
Initial version with TX, Context Propagation and Observation

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
@mp911de

### CODE REVIEW

File: spring-tx/src/main/java/org/springframework/transaction/support/DefaultTransactionTagsProvider.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		}
+		TransactionDefinition transactionDefinition = context.getTransactionDefinition();
+		Tags tags = Tags.empty();
+		if (transactionDefinition.getTimeout() > 0) {
+			tags = tags.and(TransactionObservation.LowCardinalityTags.TIMEOUT.of(String.valueOf(transactionDefinition.getTimeout())));
+		}
+		if (StringUtils.hasText(transactionDefinition.getName())) {
+			tags = tags.and(TransactionObservation.LowCardinalityTags.NAME.of(transactionDefinition.getName()));
+		}
+		return tags.and(TransactionObservation.LowCardinalityTags.TRANSACTION_MANAGER.of(ClassUtils.getQualifiedName(context.getTransactionManagerClass())),
```
Comment (marcingrzejszczak) [inline spring-tx/src/main/java/org/springframework/transaction/support/DefaultTransactionTagsProvider.java:47]: `TransactionManagerClass ` can be `null` ?

File: spring-tx/src/main/java/org/springframework/transaction/support/ObservationPlatformTransactionManager.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	@Override
+	public TransactionStatus getTransaction(TransactionDefinition definition) throws TransactionException {
+		Observation obs = this.observationRegistry.getCurrentObservation();
+		if (obs == null) {
+			obs = TransactionObservation.TX_OBSERVATION.observation(this.observationRegistry, this.context)
+					.tagsProvider(this.transactionTagsProvider)
+					.start();
+		}
+		this.observation = obs;
+		this.scope = this.observation.openScope();
```
Comment (marcingrzejszczak) [inline spring-tx/src/main/java/org/springframework/transaction/support/ObservationPlatformTransactionManager.java:67]: if there was a scope `obs != null` DO  NOT open a new scope

File: spring-tx/src/main/java/org/springframework/transaction/support/TransactionObservation.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+			@Override
+			public String getKey() {
+				return "spring.tx.timeout";
+			}
+		},
+
+		/**
+		 * Transaction name.
+		 */
+		NAME {
```
Comment (marcingrzejszczak) [inline spring-tx/src/main/java/org/springframework/transaction/support/TransactionObservation.java:105]: This is high cardinality cause name by default is a new UUID

**Comment by bclozel** [comment]
We have implemented lots of observability features already and I'm not sure we should go further ahead here with an observation per transaction call. Maybe this is similar to #30089 in the sense that we don't really need metrics/traces for this part? I'm not sure we need context propagation anyway as the transaction guarantee some context usage already.

I'm closing this issue for now and we'll revisit depending on the community requests.

**Pull request closed without merging**